### PR TITLE
Add Redshift Snapshot resource

### LINKED ISSUE #387: Add Redshift Snapshot resource
Add  Redshift Snapshot resource. Functionality to match the RDS DB Instance Snapshot (`rds-snapshot`) resources where appropriate.
Do **NOT** address tagging functionality to start with.

### PR DESCRIPTION
**Pull request by ewbankkit**
Add 'redshift-snapshot' resource
Add Redshift 'retention' and 'snapshot' actions
Add Redshift Snapshot delete action
Fix various Pylint and Flake8 issues in `redshift.py`
Fixes #387

### CODE REVIEW

File: tests/test_redshift.py [hunk 1]
```diff
@@ -63,3 +63,76 @@ def test_redshift_default_vpc(self):
...
+            'resource': 'redshift-snapshot',
+            'filters': [
+                {'type': 'value',
+                 'key': 'Encrypted',
+                 'value': False}]},
+            session_factory=session_factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+
+    # Comment out until https://github.com/capitalone/cloud-custodian/pull/374 is merged.
```
Comment (kapilt) [inline tests/test_redshift.py]: this was merged

Comment (ewbankkit) [inline tests/test_redshift.py]: Indeed. I'll uncomment.

**Comment by ewbankkit** [comment]
Uncommented. Tests pass.

**Comment by kapilt** [comment]
Thanks!, looks good

**Pull request merged**